### PR TITLE
Add 'make tests FILTER=filter'

### PR DESCRIPTION
**Pull request by uri-99**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -59,6 +59,23 @@ make build
...
+```bash
+make test
+```
+
+Or you can filter by test keyword,
+
+For example the following will display only tests related to the "Mint" functionality:
+
+```bash
+make test FILTER=Mint
```
Comment (rcatalan98) [inline README.md:75]: Is not clear if this filters for the name of the test or mod of tests. Could you provide more information and a more detailed example?

File: Makefile [hunk 2]
```diff
@@ -26,8 +26,17 @@ deploy:
 demo-local:
 	cargo run --bin local
 
+GREEN := \033[32m
+RESET := \033[0m
```
Comment (rcatalan98) [inline Makefile:30]: Should this be defined somewhere else? This could be passed by the user if it's defined here

**Review by rcatalan98 — CHANGES_REQUESTED** [review]

**Review by rcatalan98 — APPROVED** [review]

**Review by dpinones — APPROVED** [review]

File: Makefile [hunk 3]
```diff
@@ -25,9 +25,28 @@ deploy:
 	
 demo-local:
 	cargo run --bin local
+	
 
-test:
+FN  ?=
+MOD ?=
+tests:
```
Comment (rcatalan98) [inline Makefile:32]:
```suggestion
test:
```

File: README.md [hunk 4]
```diff
@@ -59,6 +59,32 @@ make build
...
+If you wish to run tests from a specific module, use the MOD parameter. For instance, to run tests within the Mint module, execute:
+
+```bash
+make test MOD=Mint
+```
+
+To focus on a specific test function, utilize the FN parameter. For instance, to run only the "test_invalid_max_tick" function:
+
+```bash
+make test FN=test_invalid_max_tick
```
Comment (rcatalan98) [inline README.md:79]: This test doesn't work. After running it I get the following error:
`scarb test | filter by test_invalid_max_tick
     Running cairo-test yas with filter=test_invalid_max_tick
testing yas ...

grep: invalid option -- P
usage: grep [-abcdDEFGHhIiJLlMmnOopqRSsUVvwXxZz] [-A num] [-B num] [-C[num]]
        [-e pattern] [-f file] [--binary-files=value] [--color=when]
        [--context[=num]] [--directories=action] [--label] [--line-buffered]
        [--null] [pattern] [file ...]
thread 'main' panicked at 'failed printing to stdout: Broken pipe (os error 32)', library/std/src/io/stdio.rs:1019:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace`

File: README.md [hunk 5]
```diff
@@ -59,6 +59,32 @@ make build
...
+```
+
+For more precise testing, you can combine both parameters. Remember, these parameters are case-sensitive.
+
+```bash
+make test MOD=Mint FN=test_invalid_max_tick
```
Comment (rcatalan98) [inline README.md:85]: This test doesn't work. I get the following error:
`

`
Also there is no` test_invalid_max_tick` function in the Mint module.

**Review by rcatalan98 — CHANGES_REQUESTED** [review]
Please make the necessary changes. Do we need to add to the README that `grep` is a requirement to run specific tests in this project?

**Comment by rcatalan98** [comment]
I just checked and it looks like you can replace all the grep with just `scarb test -f test_name` and it will only execute that specific function. No need for wrapper code anymore

**Review by rcatalan98 — APPROVED** [review]
Great work! Before merging please merge the latest changes from main.

**Pull request merged**